### PR TITLE
Fix bug with torpedos countering tml acu

### LINKED ISSUE #1164: Torpedo killing sub nuke missile
Seems like collisions between torpedos and missiles are allowed which resulted in a sub nuke missile being destroyed 1 tick after it was lauched.

On one hand it's cool that such thing can happen but in this case it was non-obvious that it actually happened (people are used with nuke missiles disappearing mysteriously but this was beta).

### PR DESCRIPTION
**Pull request by Crotalus**
Rebase of #1220

Torpedos that hit something underwater at the same time as they fire a
missiles makes it disappear

Fixes #1164

### CODE REVIEW

File: lua/sim/Projectile.lua [hunk 1]
```diff
@@ -158,38 +158,30 @@ Projectile = Class(moho.projectile_methods, Entity) {
...
-		local bp = other:GetBlueprint()
-		if bp.DoNotCollideList then
-			for k, v in pairs(bp.DoNotCollideList) do
-				if EntityCategoryContains(ParseEntityCategory(v), self) then
-					return false
-				end
-			end
-		end
+        local dnc
+        for _, p in {{self, other}, {other, self}} do
```
Comment (aeoncleanse) [inline lua/sim/Projectile.lua]: This code struck me as clever... :)

**Comment by aeoncleanse** [comment]
Restarted travis

**Comment by aeoncleanse** [comment]
eff63596cbac1eddef92737f53b3f0208ab11b94

**Comment by Exotic-Retard** [comment]
that fix is great and all but, looks like you forgot depth charges, which still do collide since they dont have the torpedo category. in fact they barely have any categories.
`        'UEF',
        'PROJECTILE',
        'ANTINAVY',`

**Pull request closed without merging**